openwisp_monitoring/monitoring/apps.py

```diff
@@ -14,3 +15,12 @@ def ready(self):
         # create influxdb database if doesn't exist yet
         create_database()
         setattr(settings, 'OPENWISP_ADMIN_SHOW_USERLINKS_BLOCK', True)
+        from ..device.models import Device
+        post_delete.connect(self.delete_related_metric,
+                            sender=Device,
+                            dispatch_uid='delete_related_metric')
+
+    @classmethod
+    def delete_related_metric(cls, instance, **kwargs):
+        from .models import Metric
+        Metric.objects.filter(object_id=instance.pk).delete()
```

Comment:
this query needs to specify the `content_type` as well, I will handle this change on my own